heareval/tasks/util/luigi.py

```diff
@@ -63,6 +77,55 @@ def stage_number(self) -> int:
             raise ValueError("Unknown requires: %s" % self.requires())
 
 
+class DownloadCorpus(WorkTask):
+    """
+    Task for downloading a dataset to a specific filename
+    """
+
+    # URL to download dataset from
+    url = luigi.Parameter()
+    outfile = luigi.Parameter()
+
+    @property
+    def name(self):
+        return type(self).__name__
+
+    def run(self):
+        download_file(self.url, os.path.join(self.workdir, self.outfile))
+        with self.output().open("w") as _:
+            pass
+
+    @property
+    def stage_number(self) -> int:
+        return 0
+
+
+class ExtractCorpus(WorkTask):
+
+    infile = luigi.Parameter()
+
+    @property
+    def name(self):
+        return type(self).__name__
+
+    def run(self):
+
+        filename, extension = os.path.split(self.infile)
+
+        # TODO -- run correct extraction given archive type
+        print(extension)
+
+        # Location of zip file to extract. Figure this out before changing
+        # the working directory.
+        corpus_zip = os.path.realpath(
+            os.path.join(self.requires().workdir, "corpus.zip")
```

Comment:
I'm confused, how did we get from self.infile to requires().workdir/corpus.zip?

Also, I think we should check the file extension and make this called ExtractZipCorpus

Reply:
We have already linked the `DownloadCorpus` to `ExtractCorpus`. I don't think we need an input to the `ExtractCorpus` since the file can be fetched from the `DownloadCorpus.workdir`.

Reply:
@khumairraj where do you see "We have already linked the DownloadCorpus to ExtractCorpus"? I don't see a self.requires for this class.

It seems to me that the correct behavior is self.requires() based upon a prerequisite DownloadCorpus.

In this case, the issue of checkpoint name is even more dangerous.

Let's say we have DownloadCorpus(file1) and DownloadCorpus(file2). They write checkpoint: 00-DownloadCorpus-file1 and 00-DownloadCorpus-file2 and if those files exist then we know that we have downloaded those file. And we have two ExtractCorpus tasks (with NO parameters but a separate DownloadCorpus requires for each). Using our existing pattern, the checkpoint will be called 01-ExtractCorpus. i.e. both tasks will have the same checkpoint name!

This is definitely making me quite worried. I'm almost wondering if we should switch to joblib which handles dependency management more cleanly I think.

@jorshi @khumairraj ^^

Reply:
The checkpoints are saved inside subdirectories of each task. For example `speech_commands-v0.0.2/00-DownloadCorpus.done`. The checkpoint for another task will be in its own subdirectory. WIll its still be a problem?

Reply:
Let's say that speech commands has two DownloadCorpus tasks. You run the pipeline and cancel it halfway through and only one DownloadCorpus command has finished. What happens then...